central/telemetry/centralclient/interceptors.go

```diff
@@ -0,0 +1,103 @@
+package centralclient
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome"
+)
+
+var (
+	trackedPaths set.FrozenSet[string]
+	ignoredPaths = []string{"/v1/ping", "/v1.PingService/Ping", "/v1/metadata", "/static/"}
+
+	clusterStatus = map[string]storage.ClusterHealthStatus_HealthStatusLabel{}
+	mux           = &sync.Mutex{}
+
+	interceptors = map[string][]phonehome.Interceptor{
+		"API Call":            {apiCall},
+		"Cluster Registered":  {postCluster},
+		"Cluster Initialized": {putCluster},
+		"roxctl":              {roxctl},
+	}
+)
+
+// Adds Path, Code and User-Agent properties to API Call events for the API
+// paths which start from the prefixes specified in the
+// rhacs.redhat.com/telemetry-apipaths central deployment annotation
+// ("*" value enables all paths) and are not in the ignoredPath list.
+func apiCall(rp *phonehome.RequestParams, props map[string]any) bool {
+	for _, ip := range ignoredPaths {
+		if strings.HasPrefix(rp.Path, ip) {
+			return false
+		}
+	}
+	if trackedPaths.Contains("*") || trackedPaths.Contains(rp.Path) {
+		props["Path"] = rp.Path
+		props["Code"] = rp.Code
+		props["User-Agent"] = rp.UserAgent
+		props["Method"] = rp.GetMethod()
+		props["Protocol"] = rp.GetProtocol()
+		return true
+	}
+	return false
+}
+
+// Adds Post Cluster call specific properties to the Post Cluster event.
+func postCluster(rp *phonehome.RequestParams, props map[string]any) bool {
```

Comment:
(same for `putCluster`): I feel like both of those would be good to be covered by unit tests.